backend/service/service_block.go

```diff
@@ -16,15 +22,108 @@ func (service *BlockService) GetModule() Module {
-	var result document.Block
+func (service *BlockService) QueryBlock(blockHeight int64) model.Block {
+	result := model.Block{}
+	result.BlockInfo = service.Query(blockHeight)
+	result.ValidatorSet = service.GetValidatorSet(blockHeight, DefaultPageNum, DefaultPageSize)
+	result.TokenFlows = Get(Tx).(*TxService).QueryTokenFlow(blockHeight, DefaultPageNum, DefaultPageSize)
+	result.Proposals = Get(Proposal).(*ProposalService).QueryProposalsByBlockHeight(blockHeight)
 
-	var err = queryOne(document.CollectionNmBlock, selector, bson.M{document.Block_Field_Height: height}, &result)
-	if err != nil {
+	return result
+}
+
+func (service *BlockService) GetValidatorSet(height int64, page, size int) model.ValidatorSet {
+	result := model.ValidatorSet{}
+	lcdValidators := lcd.ValidatorSet(height)
+	if page > 0 {
+		page = page - 1
+	}
+	items := []model.BlockValidator{}
+	for k, v := range lcdValidators.Validators {
+		if k >= page*size && k < (page+1)*size {
+			var tmp model.BlockValidator
+			tmp.Consensus = v.PubKey
+			tmp.VotingPower = v.VotingPower
+			tmp.ProposerPriority = v.ProposerPriority
+			var validatorDoc document.Validator
+			var selector = bson.M{"description.moniker": 1, "operator_address": 1}
+			err := queryOne(document.CollectionNmValidator, selector, bson.M{"consensus_pubkey": v.PubKey}, &validatorDoc)
+			if err != nil {
+				logger.Error("get validator set err", logger.String("error", err.Error()), service.GetTraceLog())
+			}
+			tmp.OperatorAddress = validatorDoc.OperatorAddress
+			tmp.Moniker = validatorDoc.Description.Moniker
+			items = append(items, tmp)
+		}
+	}
+	result.Items = items
+	result.Total = len(lcdValidators.Validators)
+	return result
+}
+
+func (service *BlockService) Query(height int64) model.BlockInfo {
+	var result model.BlockInfo
+
+	b := lcd.Block(height)
+	if b.Block.Header.Height == "" {
 		panic(types.CodeNotFound)
 	}
-	return buildBlock(result)
+
+	result.BlockHash = b.BlockMeta.BlockID.Hash
+	result.BlockHeight = b.Block.Header.Height
+
+	heightAsInt64, err := strconv.ParseInt(b.Block.Header.Height, 10, 0)
+
+	if err != nil {
+		logger.Error("block header height is not int type")
+	}
+	result.LastBlock = heightAsInt64 - 1
+	result.LastBlockHash = b.Block.Header.LastBlockID.Hash
+
+	var selector = bson.M{"description.moniker": 1, "operator_address": 1}
+	var validatorDoc document.Validator
+	err = queryOne(document.CollectionNmValidator, selector, bson.M{"proposer_addr": b.BlockMeta.Header.ProposerAddress}, &validatorDoc)
+
+	if err != nil {
+		logger.Error("query validator collection  err", logger.String("error", err.Error()), service.GetTraceLog())
+		return result
+	}
+	result.PropoperAddr = validatorDoc.OperatorAddress
+	result.PropopserMoniker = validatorDoc.Description.Moniker
+	result.Timestamp = b.BlockMeta.Header.Time
+	result.Transactions = b.BlockMeta.Header.NumTxs
```

Comment:
"result.Timestamp = b.BlockMeta.Header.Time
result.Transactions = b.BlockMeta.Header.NumTxs" ,These codes are moved to the front